components/brave_component_updater/browser/component_contents_verifier.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_COMPONENT_UPDATER_BROWSER_COMPONENT_CONTENTS_VERIFIER_H_
+#define BRAVE_COMPONENTS_BRAVE_COMPONENT_UPDATER_BROWSER_COMPONENT_CONTENTS_VERIFIER_H_
+
+#include <memory>
+
+#include "base/files/file_path.h"
+#include "base/functional/callback.h"
+
+namespace component_updater {
+
+class ContentsVerifier {
+ public:
+  virtual ~ContentsVerifier() = default;
+  virtual bool VerifyContents(const base::FilePath& relative_path,
+                              base::span<const uint8_t> contents) = 0;
+};
+
+using ContentsVerifierFactory =
+    base::RepeatingCallback<std::unique_ptr<ContentsVerifier>(
+        const base::FilePath& component_root)>;
+
+void SetContentsVerifierFactory(ContentsVerifierFactory factory);
+
+// Uses factory set above to create the verifier.
```

Comment:
I don't understand why the factory is needed, why can't we just directly create an instance of `ContentsVerifier`?

Reply:
Because we want to use it in components, but the implementation requires code from extensions.

Reply:
afaik it's fine to use extensions in components as long as it's not from chrome/browser